tools/build_api.py

```diff
@@ -161,65 +145,98 @@ def build_project(src_path, build_path, target, toolchain_name,
     except KeyError as e:
         raise KeyError("Toolchain %s not supported" % toolchain_name)
 
-    toolchain.VERBOSE = verbose
+    toolchain.config = config
```

Comment:
I don't really see the point in adding the `config` instance as a member of `toolchain`. They are different objects with different purposes. Not a blocker, just a thought.